cvmfs/sync_mediator.cc

```diff
@@ -34,13 +36,14 @@ AbstractSyncMediator::~AbstractSyncMediator() {}
 
 SyncMediator::SyncMediator(catalog::WritableCatalogManager *catalog_manager,
                            const SyncParameters *params,
-                           perf::StatisticsTemplate statistics) :
-  catalog_manager_(catalog_manager),
-  union_engine_(NULL),
-  handle_hardlinks_(false),
-  params_(params),
-  changed_items_(0)
```

Comment:
We should remove `changed_items_` as a member of `SyncMediator`.  We use `changed_items_` in `Commit()` in order to terminate the "progress bar" (the dots) in case any dots where actually printed (i.e. size of the change set >= 100).  This logic should go into the `reporter_`, e.g. with a new public method `SyncDiffReporter::CommitReport()`.